### PR TITLE
test: Coverage for NMSettingConverter

### PR DESCRIPTION
**Pull request by GregoryIvo**
This PR aims to add thorough test coverage of the NMSettingConverter for the new Networking Stack in Kura.

> **Note**: We are using the Conventional Commits convention for our pull request titles. Please take a look at the [PR title format document](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#submitting-the-changes) for the supported [types](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#type) and [scopes](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#scope).

Brief description of the PR. [e.g. Added `null` check on `object` to avoid `NullPointerException`]

**Related Issue:** This PR fixes/closes {issue number}

**Description of the solution adopted:** A more detailed description of the changes made to solve/close one or more issues. If the PR is simple and easy to understand this section can be skipped.

**Screenshots:** If applicable, add screenshots to help explain your solution

**Any side note on the changes made:** Description of any other change that has been made, which is not directly linked to the issue resolution [e.g. Code clean up/Sonar issue resolution]

### CODE REVIEW

**Comment by GregoryIvo** [comment]
@mattdibi  ready for Review!

image

**Comment by MMaiero** [comment]
Sonar check succeeds as well as the Jenkins build of the code. The problem seems to be related to a bad communication in Jenkins to get the sonar cloud status as in the following image:
Screenshot 2023-02-16 at 10 55 29

Waiting for the review by @mattdibi before merging

File: kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java
Comment (mattdibi) [inline kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java]: This is unnecessary:

```suggestion
        thenResultingMapContains("channel", new UInt32(10));
```

File: kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java
Comment (mattdibi) [inline kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java]: Again unnecessary complication:

```suggestion
        thenResultingMapContains("channel", new UInt32(10));
```

**Review by mattdibi — APPROVED** [review]
image

**Pull request merged**
